packages/@n8n/config/src/configs/database.config.ts

```diff
@@ -75,6 +75,10 @@ class PostgresConfig {
 	@Env('DB_POSTGRESDB_POOL_SIZE')
 	poolSize: number = 2;
 
+	/** Postgres connection timeout (ms) */
+	@Env('DB_POSTGRESDB_CONNECTION_TIMEOUT')
+	connectionTimeoutMs: number = 1000;
```

Comment:
1s is a very short default value. Something like 10-30s would make more sense. Also remember to update docs

Reply:
According to Adi already 5s was too long? https://github.com/n8n-io/n8n/pull/10670#discussion_r1744020287

The docs PR is already linked above: https://github.com/n8n-io/n8n-docs/pull/2452

Reply:
typeorm docs say
> The milliseconds before a timeout occurs during the initial connection to the postgres server.

So this is only the timeout for opening the connection. If that takes longer than 1 second, we might have many other issues in the setup. Setting this to 30s might not be a great idea.

Reply:
`typeorm` passes that into `connectionTimeoutMillis` for `pg` lib's `Pool`. So it includes the whole TCP connection opening, TLS handshake, PG wire protocol including authentication etc. I fear users might run into bunch of issues if the default is very tight

Reply:
> it includes the whole TCP connection opening, TLS handshake, PG wire protocol including authentication

all of that should not take more than a second, even under heavy load. If it does, there are fundamental infrastructure issues that need to be addressed, instead of letting all of this take 30 seconds before timing out.

Reply:
Well, the question is do we really want to support a scenario like this with large latencies by default, or if anyone with a setup like this should have to manually set `DB_POSTGRESDB_CONNECTION_TIMEOUT` to a much higher value?

IMO, we should officially only support colocated databases, to avoid having to support all the other issues that large latencies might cause.